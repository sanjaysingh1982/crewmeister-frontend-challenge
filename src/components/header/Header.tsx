import { MDBNavbar, MDBNavbarBrand, MDBContainer } from 'mdbreact';

const Header = () => {
    return (
        <header>
            <MDBNavbar color="black" dark expand="md">
                <MDBContainer>
                    <MDBNavbarBrand>
                        <strong>Crewmeister Frontend Challenge</strong>
                    </MDBNavbarBrand>
                </MDBContainer>
            </MDBNavbar>
        </header>
    )
};

export default Header;