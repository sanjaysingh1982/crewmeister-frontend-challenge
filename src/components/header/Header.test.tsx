import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Header from './Header';

const setup = (props = {}, state = null) => {
    const defaultProps = {};
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<Header {...setUpProps}/>);
    const tree = () => renderer.create(<Header {...setUpProps}/>).toJSON();
    if (state) wrapper.setState(state);
    return { wrapper, props: setUpProps, tree };
};

describe('Header component', () => {
    it('should match with snapshot', () => {
        const { tree } = setup();
        expect(tree()).toMatchSnapshot();
    });

    it('Should contain header tag', () => {
        const { wrapper } = setup();
        expect(wrapper.find('header').length).toBe(1);
    });
});