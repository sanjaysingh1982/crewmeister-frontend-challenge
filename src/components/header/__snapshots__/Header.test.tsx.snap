// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match with snapshot 1`] = `
<header>
  <nav
    className="navbar-dark black navbar navbar-expand-md"
    data-test="navbar"
    role="navigation"
  >
    <div
      className="container"
      data-test="container"
    >
      <div
        className="navbar-brand"
        data-test="navbar-brand"
      >
        <strong>
          Crewmeister Frontend Challenge
        </strong>
      </div>
    </div>
  </nav>
</header>
`;
